BWAPI4J/src/main/java/org/openbw/bwapi4j/unit/Lair.java

```diff
@@ -20,48 +20,51 @@
 
 package org.openbw.bwapi4j.unit;
 
+import org.openbw.bwapi4j.type.UnitCommandType;
+import org.openbw.bwapi4j.type.UnitType;
+import org.openbw.bwapi4j.type.UpgradeType;
+
 import static org.openbw.bwapi4j.type.UnitCommandType.Morph;
 import static org.openbw.bwapi4j.type.UnitType.Zerg_Hive;
 
-import org.openbw.bwapi4j.type.UnitType;
-import org.openbw.bwapi4j.type.UpgradeType;
+public class Lair extends ZergResourceDepotImpl {
```

Comment:
There seems to be only formatting changes with this file. This formatting does not match the rest of the codebase.

Reply:
`isReadyForResources()` was removed and Lair now extends from a `ZergResourceDepotImpl` abstract. I believe not having `Lair` extend `Hatchery` better reflects the game. For example, somebody might want to check if there is a `Hatchery` to research `Burrow` and use their `Lair` for other upgrades - previously, checking `instanceof Hatchery` on our `Lair would return true.

Reply:
Yes, sorry my mistake.

Reply:
I don't agree here. I'm actually using the `Lair` is a `Hatch` mechanic in my bot, as it makes it easier to check if I already have a `Lair` although it is updated to a `Hive` already.
As we discussed on Discord, both ways have their pros and contras.

Currently you'll have to check if you supposed `Hatchery` is actually a `Lair` or a `Hive`.
After the change, you'd have to check if you've got a `Lair` or a `Hive`. You only move the check to another position.

Unless there's an additional benefit, this is just toggling code back and forth.

Reply:
The proposed scheme would have 4 different types of checks that could be accomplished with a single comparison (If something is a `Hatchery`, `Lair`, `Hive`, or all three) - no second check necessary. If one wants to select two types, that can be accomplished with a second comparison. 

With the current arrangement, with a single check, one can filter a `Unit` down to all a `Hatchery` || `Lair` || `Hive`, a `Lair` || `Hive`, or a `Hive`. Further necking down requires a second check. 

More importantly, I think it is not intuitive to check if a `Unit` is a `Lair` and have it return true for a `Hive`. When I ask if a `Unit` is a `Hatchery`, it's not obvious at all `Lair` and `Hive` qualify as a `Hatchery`

Reply:
We're repeating our discord :)
This is not at all an objective argument we're having here. It's just "taste".
I for example find it very intuitive that a `Hive`  *isa* `Lair`- it's the same way it is presented to a player in-game.

There's no *objective* benefit doing it either way. And I believe every change *breaking API* should.